bloom-housing/bloom
# Listings CSV

## Issue Overview

This PR addresses #3732 


## Description

This PR adds the listings csv export functionality to the backend. It is a complete rewrite from the old csv export functionality. This first writes the listings and units csvs and then archives them before sending as the response. It is a branch of https://github.com/bloom-housing/bloom/pull/3771, so if you haven't reviewed that, please do so first.

That PR doesn't actually zip up the file, so we may want to update that to get the file size even smaller, like here.

Updates for access control need to added as well as tests.

## How Can This Be Tested/Reviewed?

1. pull down this branch
2. there are new dependencies, so please run `yarn`
3. open up "backend_new/prisma/constants.ts" and edit to your desired seed number, for testing this
4. `cd backend_new`, seed the db with `yarn setup:dev` and start the new backend `yarn dev`
5. you'll need to be logged in, so through your client (I'm using thunder client for vs code - https://www.thunderclient.com/), run a post request on http://localhost:3101/auth/login with the following in the body JSON:
```
{
  "email": "admin@example.com",
  "password": "abcdef"
}
```
Copy the access token for the next step
6. make a get request to http://localhost:3101/listings/csv with the access token as the Bearer token
7. you should see the option to save the file, do so to open and view
<img width="1169" alt="Screenshot 2024-01-10 at 6 07 30 PM" src="https://github.com/bloom-housing/bloom/assets/4401157/56a9cc2d-d92e-451c-a37e-0543d1f36ba2">


## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
